### PR TITLE
Added new command: export-requirements, import-requirements, install-requirements

### PR DESCRIPTION
**Pull request by MeirShpilraien**

### CODE REVIEW

File: gears_cli/__main__.py [hunk 1]
```diff
@@ -67,7 +68,29 @@ def print_res(res, res_id):
...
+@click.option('--password', default=None, help='Redis password')
+@click.option('--requirements-file', default=None, help='Path to requirements.txt file')
+@click.argument('requirements', nargs=-1, type=click.UNPROCESSED)
+def install_requirements(host, port, password, requirements_file, requirements):
+    r = create_connection(host, port, password);
+
+    requirements = [a for a in requirements]
+
+    if requirements_file is not None:
+        with open(requirements, 'r') as f:
```
Comment (rafie) [inline gears_cli/__main__.py]: requirements_file

Comment (MeirShpilraien) [reply gears_cli/__main__.py]: Isn't that what I wrote?

File: gears_cli/__main__.py [hunk 2]
```diff
@@ -67,7 +68,29 @@ def print_res(res, res_id):
...
+@gears_cli.command(help='Install give requirements')
+@click.option('--host', default='localhost', help='Redis host to connect to')
+@click.option('--port', default=6379, type=int, help='Redis port to connect to')
+@click.option('--password', default=None, help='Redis password')
+@click.option('--requirements-file', default=None, help='Path to requirements.txt file')
+@click.argument('requirements', nargs=-1, type=click.UNPROCESSED)
+def install_requirements(host, port, password, requirements_file, requirements):
+    r = create_connection(host, port, password);
+
+    requirements = [a for a in requirements]
```
Comment (rafie) [inline gears_cli/__main__.py]: `requirements = list(requirements)` ?

File: gears_cli/__main__.py [hunk 3]
```diff
@@ -109,35 +132,35 @@ def run(host, port, password, requirements, filepath, extra_args):
...
-
-    metaData = {}
-    for i in range(0, len(metaDataValues), 2):
-        key = metaDataValues[i]
-        value = metaDataValues[i + 1]
+def extract_metadata(meta_data_reply):
+    meta_data = {}
+    for i in range(0, len(meta_data_reply), 2):
+        key = meta_data_reply[i]
+        value = meta_data_reply[i + 1]
```
Comment (rafie) [inline gears_cli/__main__.py]: We can use something of the sort of:
```
def decode_utf(d):
    if isinstance(d, dict):
        return {k.decode('utf-8'): v.decode('utf-8') for k, v in d.items()}
    if isinstance(d, list):
        return [x.decode('utf-8') for x in d]
    if isinstance(d, bytes):
        return d.decode('utf-8')
    return d.decode('utf-8') # let it fail
```

Comment (MeirShpilraien) [reply gears_cli/__main__.py]: I thought about this but then you need to call it recursively for each element inside the dict/list (if we want really a generic solution). So I thought for this specific case its enough.

File: gears_cli/__main__.py (hunk 2 above)
Comment (MeirShpilraien) [reply gears_cli/__main__.py]: I think it will create list with one element that contains the requirements no?

File: gears_cli/__main__.py (hunk 1 above)
Comment (rafie) [reply gears_cli/__main__.py]: open(requirements, 'r')

Comment (MeirShpilraien) [reply gears_cli/__main__.py]: Nice catch

File: gears_cli/__main__.py (hunk 2 above)
Comment (rafie) [reply gears_cli/__main__.py]: donno. `[...]` looks like a materialization of an iterator (it gets an iterator and returns a concrete list. but maybe i don't read this correctly.

File: gears_cli/__main__.py (hunk 3 above)
Comment (rafie) [reply gears_cli/__main__.py]: Just playing with it:
```
def decode_utf(d):
    if isinstance(d, dict):
        return {k.decode('utf-8'): decode_utf(v) for k, v in d.items()}
    if isinstance(d, list):
        return [decode_utf(x) for x in d]
    return d.decode('utf-8')
```
And we may optimize it by specifically asking about bytes before recursing.

**Review by rafie — APPROVED** [review]

**Review by rafie — APPROVED** [review]

**Pull request merged**